sdk/mixedreality/azure-mixedreality-authentication/src/main/java/com/azure/mixedreality/authentication/JsonWebToken.java

```diff
@@ -43,18 +42,17 @@ public static OffsetDateTime retrieveExpiration(String jwtValue) {
 
         byte[] jwtPayloadDecodedData = Base64.getDecoder().decode(jwtPayloadEncoded);
 
-        JsonNode rootNode;
-        try {
-            rootNode = MAPPER.readTree(jwtPayloadDecodedData);
-        } catch (IOException exception) {
-            return null;
-        }
+        try (JsonReader jsonReader = JsonProviders.createReader(jwtPayloadDecodedData)) {
+            Map<String, Object> jsonTree = jsonReader.readMap(JsonReader::readUntyped);
+            Object exp = jsonTree.get("exp");
 
-        if (!rootNode.has("exp")) {
+            if (exp == null) {
+                return null;
+            } else {
+                return OffsetDateTime.ofInstant(Instant.ofEpochSecond(Long.parseLong(exp.toString())), ZoneOffset.UTC);
+            }
```

Comment:
If we think this will be a hot path, could make a check for `exp instanceof long` before calling `toString` and `Long.parseLong` on it

Reply:
For some reason, `readUntyped()` was creating exp as a `String`, causing a `ClassCastException` when attempting to cast top `long` directly.